auth/auth.go

```diff
@@ -32,6 +42,26 @@ func New(
 	pgStore postgres.PersistentStore,
 	logger telemetry.Logger,
 ) Authentication {
-	a := &auth{store: s, c: c, pgStore: pgStore, logger: logger}
+
+	githubOAuth := &oauth2.Config{
+		ClientID:     c.OAuth.Github.ClientID,
+		ClientSecret: c.OAuth.Github.ClientSecret,
+		Endpoint:     github.Endpoint,
+		RedirectURL:  fmt.Sprintf("%s/auth/github/callback", c.Endpoint()),
+		Scopes:       []string{"user:email"},
+	}
+
+	ghClient := gh.NewClient(nil)
+
+	a := &auth{
+		store:           s,
+		c:               c,
+		pgStore:         pgStore,
+		logger:          logger,
+		github:          githubOAuth,
+		ghClient:        ghClient,
+		oauthStateToken: "skljdfkljsdjfklj",
```

Comment:
state token can not be hardcoded and should be rotated dynamically.

Reply:
https://auth0.com/docs/secure/attack-protection/state-parameters

Reply:
https://echo.labstack.com/middleware/csrf/

Reply:
great finds, I totally forgot about this one. @guacamole do you want to take this one on? just add a random small string generator or maybe just use `uuid.NewString()` instead of this token?

Reply:
Sure! I will work on it